common/src/utils/environment.ts

```diff
@@ -33,9 +35,64 @@ export const getOverrides = <T extends Partial<Record<string, unknown>>>(
 
 export type Environment = "prod" | "dev" | "staging";
 
-export const getEnvironment = (): Environment =>
-  process.env.RENDER_SERVICE_ID
-    ? process.env.RENDER_SERVICE_ID.includes("pr")
-      ? "staging"
-      : "prod"
-    : "dev";
+export const getEnvironment = (): Environment => {
+  if (process.env.RENDER && process.env.RENDER_SERVICE_ID) {
+    return process.env.RENDER_SERVICE_ID.includes("pr") ? "staging" : "prod";
+  }
+  if (process.env.GCP && process.env.GCP_REVISION_ID) {
+    return process.env.GCP_REVISION_ID.includes("pr") ? "staging" : "prod";
+  }
+  return "dev";
+};
+
+type EnvironmentVariables = {
+  environment: Environment;
+  branch: string;
+  commit: string;
+  // frontend / backend-mainnet / backend-testnet
+  serviceName: string;
+  // production / pr-xxx
+  revisionId: string;
+  // vm-1 / vm-2
+  instanceId: string;
+};
+
+export const getEnvironmentVariables = async (
+  localServiceName: string
+): Promise<EnvironmentVariables | undefined> => {
+  if (process.env.RENDER || process.env.GCP) {
+    const environment = getEnvironment();
+    if (process.env.RENDER) {
+      return {
+        branch: process.env.RENDER_GIT_BRANCH || "unknown",
+        commit: process.env.RENDER_GIT_COMMIT || "unknown",
+        serviceName: process.env.RENDER_SERVICE_NAME || "unknown",
+        revisionId: process.env.RENDER_SERVICE_ID || "unknown",
+        instanceId: process.env.RENDER_INSTANCE_ID || "unknown",
+        environment,
+      };
+    }
+    if (process.env.GCP) {
+      return {
+        branch: process.env.BRANCH || "unknown",
```

Comment:
@ecp88 Let me know when variable names will be known

Reply:
They are available in https://github.com/near/near-explorer/pull/1162/files
lets review and merge that pr.